### ISSUE TITLE
Case where control-flow merge is not handled properly

### ISSUE THREAD

**Issue by msridhar** [kelloggm/typesafe-builder-checker#30]
For the following code, we get an unexpected error from the checker:

```java
class SimpleInferenceMerge {
    void build(@CalledMethods({"a, b"}) SimpleInferenceMerge this) { }
    void a() { }
    void b() {}
    void c() {}
    static void doStuffCorrectMerge(boolean b) {
        SimpleInferenceMerge s = new SimpleInferenceMerge();
        if (b) {
            s.a();
            s.b();
        } else {
            s.b();
            s.a();
            s.c();
        }
        s.build();
    }
}
```
The checker reports a `method.invocation.invalid` error at the `build()` call, but `a` and `b` were invoked on both incoming control-flow paths.  Maybe something is wrong with our LUB computation?  The test is pushed on the `control-flow-merge` branch.

**Comment by kelloggm** [comment]
There is a typo in the test case that caused it to fail: `@CalledMethods({"a, b"})` should be `@CalledMethods({"a", "b"})`. I'll make a PR with the corrected test.
